src/EventStore.Client/EventStoreClientSettings.ConnectionString.cs

```diff
@@ -0,0 +1,246 @@
+		/// <returns></returns>
+		public static EventStoreClientSettings Create(string connectionString) {
+			return ConnectionStringParser.Parse(connectionString);
+		}
+
+		private static class ConnectionStringParser {
+			private const string SchemeSeparator = "://";
+			private const string UserInfoSeparator = "@";
+			private const string Colon = ":";
+			private const string Slash = "/";
+			private const string Comma = ",";
+			private const string Ampersand = "&";
+			private const string Equal = "=";
+			private const string QuestionMark = "?";
+			private static readonly string[] Schemes = { "esdb" };
+			private static readonly int DefaultPort = EventStoreClientConnectivitySettings.Default.Address.Port;
+			private static readonly bool DefaultUseTls = true;
+
+			private static readonly Dictionary<string, Type> SettingsType = new Dictionary<string, Type> (StringComparer.InvariantCultureIgnoreCase) {
+				{"ConnectionName", typeof(string)},
+				{"MaxDiscoverAttempts", typeof(int)},
+				{"DiscoveryInterval", typeof(int)},
+				{"GossipTimeout", typeof(int)},
+				{"NodePreference", typeof(string)},
+				{"Tls", typeof(bool)},
+				{"TlsVerifyCert", typeof(bool)},
+				{"OperationTimeout", typeof(int)},
+				{"ThrowOnAppendFailure", typeof(bool)}
+			};
+
+			public static EventStoreClientSettings Parse(string connectionString) {
+				var currentIndex = 0;
+				var schemeIndex = connectionString.IndexOf(SchemeSeparator, currentIndex, StringComparison.Ordinal);
+				if (schemeIndex == -1)
+					throw new NoSchemeException();
+				var scheme = ParseScheme(connectionString.Substring(0, schemeIndex));
+
+				currentIndex = schemeIndex + SchemeSeparator.Length;
+				var userInfoIndex = connectionString.IndexOf(UserInfoSeparator, currentIndex, StringComparison.Ordinal);
+				(string user, string pass) userInfo = (null, null);
+				if (userInfoIndex != -1) {
+					userInfo = ParseUserInfo(connectionString.Substring(currentIndex, userInfoIndex - currentIndex));
+					currentIndex = userInfoIndex + UserInfoSeparator.Length;
+				}
+
+				var slashIndex = connectionString.IndexOf(Slash, currentIndex, StringComparison.Ordinal);
+				if (slashIndex == -1)
+					throw new ConnectionStringParseException("The connection string must contain a forward slash (/) after specifying the hosts");
+
+				var hosts = ParseHosts(connectionString.Substring(currentIndex, slashIndex - currentIndex));
+				currentIndex = slashIndex + Slash.Length;
+
+				var questionMarkIndex = connectionString.IndexOf(QuestionMark, currentIndex, StringComparison.Ordinal);
+				if ((questionMarkIndex == -1 && slashIndex != connectionString.Length - Slash.Length)
+				    || (questionMarkIndex != -1 && questionMarkIndex != currentIndex))
+					throw new ConnectionStringParseException($"The connection string may contain only a question mark (?) after specifying the hosts and a forward slash (/) but the following character was found: '{connectionString[currentIndex]}'");
```

Comment:
I think this error might be a bit confusing, is there a way we could reword it?
The best I can come up with right now is:

```suggestion
					throw new ConnectionStringParseException($"The connection string must contain a question mark (?) following the hosts if more arguments are specified. The following character was found instead: '{connectionString[currentIndex]}'");
```

Reply:
@hayley-jean  this looks better, I've committed the changes, thanks